includes/libs/CSSMin.php

```diff
@@ -50,176 +54,474 @@ class CSSMin {
+
+				// Check for global @embed comment and remove it. Allow other comments to be present
+				// before @embed (they have been replaced with placeholders at this point).
+				$embedAll = false;
+				$rule = preg_replace(
+					'/^((?:\s+|' .
+						CSSMin::PLACEHOLDER .
+						'(\d+)x)*)' .
+						CSSMin::EMBED_REGEX .
+						'\s*/',
+					'$1',
+					$rule,
+					1,
+					$embedAll
+				);
+
+				// Build two versions of current rule: with remapped URLs
+				// and with embedded data: URIs (where possible).
+				$pattern = '/(?P<embed>' . CSSMin::EMBED_REGEX . '\s*|)' . self::getUrlRegex() . '/';
+
+				$ruleWithRemapped = preg_replace_callback(
+					$pattern,
+					function ( $match ) use ( $local, $remote ) {
+						self::processUrlMatch( $match );
+
+						$remapped = CSSMin::remapOne( $match['file'], $match['query'], $local, $remote, false );
+						return CSSMin::buildUrlValue( $remapped );
+					},
+					$rule
+				);
+
+				if ( $embedData ) {
+					// Remember the occurring MIME types to avoid fallbacks when embedding some files.
+					$mimeTypes = [];
+
+					$ruleWithEmbedded = preg_replace_callback(
+						$pattern,
+						function ( $match ) use ( $embedAll, $local, $remote, &$mimeTypes ) {
+							self::processUrlMatch( $match );
+
+							$embed = $embedAll || $match['embed'];
+							$embedded = CSSMin::remapOne(
+								$match['file'],
+								$match['query'],
+								$local,
+								$remote,
+								$embed
+							);
+
+							$url = $match['file'] . $match['query'];
+							$file = "{$local}/{$match['file']}";
+							if (
+								!self::isRemoteUrl( $url ) && !self::isLocalUrl( $url )
+								&& file_exists( $file )
+							) {
+								$mimeTypes[ CSSMin::getMimeType( $file ) ] = true;
+							}
+
+							return CSSMin::buildUrlValue( $embedded );
+						},
+						$rule
 					);
+
+					// Are all referenced images SVGs?
+					$needsEmbedFallback = $mimeTypes !== [ 'image/svg+xml' => true ];
 				}
-				// Move the offset to the end of the match, leaving it alone
-				$offset = $match[0][1] + strlen( $match[0][0] ) + $lengthIncrease;
-				continue;
+
+				if ( !$embedData || $ruleWithEmbedded === $ruleWithRemapped ) {
+					// We're not embedding anything, or we tried to but the file is not embeddable
+					return $ruleWithRemapped;
+				} elseif ( $embedData && $needsEmbedFallback ) {
+					// Build 2 CSS properties; one which uses a data URI in place of the @embed comment, and
+					// the other with a remapped and versioned URL with an Internet Explorer 6 and 7 hack
+					// making it ignored in all browsers that support data URIs
+					return "$ruleWithEmbedded;$ruleWithRemapped!ie";
+				} else {
+					// Look ma, no fallbacks! This is for files which IE 6 and 7 don't support anyway: SVG.
+					return $ruleWithEmbedded;
+				}
+			}, $source );
+
+		// Re-insert comments
+		$pattern = '/' . CSSMin::PLACEHOLDER . '(\d+)x/';
+		$source = preg_replace_callback( $pattern, function ( $match ) use ( &$comments ) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$source". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)